src/ptyfwd/ptyfwd-tool.c

```diff
@@ -0,0 +1,207 @@
+        } else if (rcode < 0) {
+                (void) sd_event_exit(e, EXIT_FAILURE);
+                return log_error_errno(rcode, "Error on PTY forwarding logic: %m");
+        }
+
+        return 0;
+}
+
+static int helper_on_exit(sd_event_source *s, const siginfo_t *si, void *userdata) {
+        /* Add 128 to signal exit statuses to mimick shells. */
+        return sd_event_exit(sd_event_source_get_event(s), si->si_status + (si->si_code == CLD_EXITED ? 0 : 128));
+}
+
+static int run(int argc, char *argv[]) {
+        _cleanup_(sd_event_unrefp) sd_event *event = NULL;
+        _cleanup_close_ int pty_fd = -EBADF, peer_fd = -EBADF;
+        _cleanup_(pty_forward_freep) PTYForward *forward = NULL;
+        _cleanup_(pidref_done) PidRef pidref = PIDREF_NULL;
+        _cleanup_(sd_event_source_unrefp) sd_event_source *exit_source = NULL;
+        int r;
+
+        log_setup();
+
+        assert_se(sigprocmask_many(SIG_BLOCK, /*ret_old_mask=*/ NULL, SIGCHLD) >= 0);
+
+        r = parse_argv(argc, argv);
+        if (r <= 0)
+                return r;
+
+        _cleanup_strv_free_ char **l = strv_copy(argv + optind);
+        if (!l)
+                return log_oom();
+
+        r = sd_event_default(&event);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get event loop: %m");
+
+        (void) sd_event_set_signal_exit(event, true);
+
+        pty_fd = openpt_allocate(O_RDWR|O_NOCTTY|O_NONBLOCK|O_CLOEXEC, /*ret_peer=*/ NULL);
+        if (pty_fd < 0)
+                return log_error_errno(pty_fd, "Failed to acquire pseudo tty: %m");
+
+        peer_fd = pty_open_peer(pty_fd, O_RDWR|O_NOCTTY|O_CLOEXEC);
+        if (peer_fd < 0)
+                return log_error_errno(peer_fd, "Failed to open pty peer: %m");
+
+        if (!arg_quiet)
+                log_info("Press ^] three times within 1s to disconnect TTY.");
+
+        r = pty_forward_new(event, pty_fd, /*flags=*/ 0, &forward);
+        if (r < 0)
+                return log_error_errno(r, "Failed to create PTY forwarder: %m");
+
+        if (!isempty(arg_background)) {
+                r = pty_forward_set_background_color(forward, arg_background);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to set background color: %m");
+        }
+
+        if (shall_set_terminal_title() && !isempty(arg_title)) {
+                r = pty_forward_set_title(forward, arg_title);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to set title: %m");
+        }
+
+        pty_forward_set_handler(forward, pty_forward_handler, NULL);
+
+        r = pidref_safe_fork_full(
+                        "(sd-ptyfwd)",
+                        (int[]) { peer_fd, peer_fd, peer_fd },
+                        /*except_fds=*/ NULL,
+                        /*n_except_fds=*/ 0,
+                        /*flags=*/ FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGTERM|FORK_LOG|FORK_REARRANGE_STDIO|FORK_SETSID|FORK_TIOCSCTTY,
```

Comment:
I'd really like proper spaces...

Reply:
Let's codify this in CODING_STYLE.md first, we do all kinds of formatting for these kind of comments atm, let's standardize on one before trying to convert everyone to one style

Reply:
At least spaces before/after `*` is something everyone else agrees on.